roles/docker/tasks/cleanup.yml

```diff
@@ -1,6 +1,14 @@
 ---
 # This play contains tasks for cleaning up docker
 
+- name: remove all containers
+  shell: unset DOCKER_HOST && docker rm -f -v $(docker ps -a -q)
```

Comment:
Yes, it removes all containers, including swarm. As docker cleanup is run after swarm cleanup, this is fine. Uninstaller currently does not uninstall docker or any component. Cleanup only stops services. My guess is it was designed this way to allow multiple install/upgrades without having to deal with issues like the docker upgrade here.
We do need a set of uninstall tasks for all roles, to remove files, uninstall binaries etc. Thats a separate issue.